trin-core/src/utp/stream.rs

```diff
@@ -606,6 +623,7 @@ impl UtpStream {
             let mut packet = Packet::with_payload(chunk);
             packet.set_seq_nr(self.seq_nr);
             packet.set_ack_nr(self.ack_nr);
+            packet.set_wnd_size(WINDOW_SIZE.wrapping_sub(self.cur_window));
```

Comment:
When we allow an underflow like this, it seems worth at least a short comment to explain why we're not worrying about it. Something like:

> If `WND_SIZE` underflows to end up higher than we intended, then we would be advertising for the peer to send us more data than we would theoretically prefer. That's not a fatal situation, so we don't protect against it or warn.

Reply:
Actually, I changed this to [saturated_sub](https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_sub) which advertises a remote window size of 0, instead of allowing underflow. I think this is better than asking for more data than we would theoretically prefer.